saleor/graphql/notifications/tests/test_external_notification_query.py

```diff
@@ -0,0 +1,508 @@
+        sendgrid_email_plugin,
+        product_with_two_variants,
+        permission_manage_products,
+    ):
+        ids = [
+            to_global_id(ProductVariant.__name__, pk)
+            for pk in product_with_two_variants.variants.values_list("pk", flat=True)
+        ]
+        self._common_part(
+            sendgrid_plugin_notify,
+            user_email_plugin_notify,
+            admin_email_plugin_notify,
+            webhook_plugin_notify,
+            staff_api_client,
+            external_notification_trigger_query,
+            sendgrid_email_plugin,
+            ids,
+            permission_manage_products,
+        )
+
+    @patch("saleor.plugins.webhook.plugin.WebhookPlugin.notify")
+    @patch("saleor.plugins.admin_email.plugin.AdminEmailPlugin.notify")
+    @patch("saleor.plugins.user_email.plugin.UserEmailPlugin.notify")
+    @patch("saleor.plugins.sendgrid.plugin.SendgridEmailPlugin.notify")
+    def test_for_single_product_variant(
+        self,
+        sendgrid_plugin_notify,
+        user_email_plugin_notify,
+        admin_email_plugin_notify,
+        webhook_plugin_notify,
+        staff_api_client,
+        external_notification_trigger_query,
+        sendgrid_email_plugin,
+        product_with_single_variant,
+        permission_manage_products,
+    ):
+        ids = [
+            to_global_id(ProductVariant.__name__, pk)
+            for pk in product_with_single_variant.variants.values_list("pk", flat=True)
+        ]
+        self._common_part(
+            sendgrid_plugin_notify,
+            user_email_plugin_notify,
+            admin_email_plugin_notify,
+            webhook_plugin_notify,
+            staff_api_client,
+            external_notification_trigger_query,
+            sendgrid_email_plugin,
+            ids,
+            permission_manage_products,
+        )
+
+    @patch("saleor.plugins.webhook.plugin.WebhookPlugin.notify")
+    @patch("saleor.plugins.admin_email.plugin.AdminEmailPlugin.notify")
+    @patch("saleor.plugins.user_email.plugin.UserEmailPlugin.notify")
+    @patch("saleor.plugins.sendgrid.plugin.SendgridEmailPlugin.notify")
+    def test_for_customers(
+        self,
+        sendgrid_plugin_notify,
+        user_email_plugin_notify,
+        admin_email_plugin_notify,
+        webhook_plugin_notify,
+        staff_api_client,
+        external_notification_trigger_query,
+        sendgrid_email_plugin,
+        staff_users,
+        permission_manage_users,
+    ):
+        ids = [to_global_id(User.__name__, user.id) for user in staff_users]
+        self._common_part(
+            sendgrid_plugin_notify,
+            user_email_plugin_notify,
+            admin_email_plugin_notify,
+            webhook_plugin_notify,
+            staff_api_client,
+            external_notification_trigger_query,
+            sendgrid_email_plugin,
+            ids,
+            permission_manage_users,
+        )
+
+    def _common_part(
```

Comment:
If all tests have "the common part", shouldn't they be rewritten with the usage of `pytest.mark.parametrize`?